src/com/yashu/interview/map/linkedhashset/HashSet.java

```diff
@@ -0,0 +1,218 @@
+package HashSet;
+import java.util.InputMismatchException;
+public class HashSet<T> implements OrderedSet<T>, Container<T>{
+
+	private	Node<T>[] array;
+	private ArrayIndexCalculator indexCalculator;
+	private LinkedList<T> linkedList;
+	private int count = 0;
+
+
+	/**
+	 * Parameterized constructor
+	 * @param initialSize - Initial size of the set
+	 */
+	@SuppressWarnings("unchecked")
+	public HashSet(int initialSize){
+		array = new Node[initialSize];
+		indexCalculator = new ArrayIndexCalculator();
+		linkedList = new LinkedList<T>();
+	}
+	 
+	// Package visible only - for testing purposes
+	@SuppressWarnings("unchecked")
+	HashSet(int initialSize, ArrayIndexCalculator indexCalculator) {
+		array = new Node[initialSize];
+		this.indexCalculator = indexCalculator;
+		linkedList = new LinkedList<T>();
+	}
+	// Returns the size of the Array.
+	public int getLengthOfArray() {
+		return array.length;
+	}
+
+	@SuppressWarnings("unchecked")
+	private void createNewArray(int length) {
+		array = new Node[length];
+	}
+	
+	//Iterator
+	private class LinkedHashSetIterator implements MyIterator<T>{
+		private Node<T> temp ;
+		
+		public LinkedHashSetIterator(){
+		temp = null;
+		}
+		@Override
+		public boolean hasNext() {
+			return temp!=linkedList.getTail();
+		}
+
+		@Override
+		public Object next() {
+			if(temp == null){
+				temp = linkedList.getHead() ;
+				return temp.getData();
+			}
+			temp = temp.getNext();
+			if(this.hasNext()){
+				return temp.getData();
+			}
+			return temp.getData();
```

Comment:
What is the purpose of  if() condition on line 58. It is redundant. You can remove it. 

What happens if I call next() even if hasNext() returns null? It will throw a null pointer exception. 
You should through an exception (NoSuchElementException) as defined in Iterator interface.